### ISSUE TITLE
Magento 2.2.1+: DataProvider constructor PHPDoc is missing string type

### ISSUE THREAD

**Issue by VincentMarmiesse** [staempfli/magento2-code-generator#9]
Hello,
When creating a `crud` template, the PHPDoc of the DataProvider `__construct` does not specify the `string` type.

Actual constructor:
```
/**
     * DataProvider constructor.
     * @param $name
     * @param $primaryFieldName
     * @param $requestFieldName
     * @param Collection $collection
     * @param FilterPool $filterPool
     * @param array $meta
     * @param array $data
     */
    public function __construct(
        $name,
        $primaryFieldName,
        $requestFieldName,
        Collection $collection,
        FilterPool $filterPool,
        array $meta = [],
        array $data = []
    ) {
        parent::__construct($name, $primaryFieldName, $requestFieldName, $meta, $data);
        $this->collection = $collection;
        $this->filterPool = $filterPool;
    }
```

Expected:
```
/**
     * DataProvider constructor.
     * @param string $name
     * @param string $primaryFieldName
     * @param string $requestFieldName
     * @param Collection $collection
     * @param FilterPool $filterPool
     * @param array $meta
     * @param array $data
     */
    public function __construct(
        $name,
        $primaryFieldName,
        $requestFieldName,
        Collection $collection,
        FilterPool $filterPool,
        array $meta = [],
        array $data = []
    ) {
        parent::__construct($name, $primaryFieldName, $requestFieldName, $meta, $data);
        $this->collection = $collection;
        $this->filterPool = $filterPool;
    }
```
  
From Magento 2.2.1, the DI is looking at the constructor comments and compilation fails if it does'nt have the types with the error `Incompatible argument type: Required type: string. Actual type: \Vendor\Module\Ui\Component\Form\Entity\name;`

**Comment by jalogut** [comment]
Hi @VincentMarmiesse 
Thanks for pointing that out. We have not used this tool with Magento 2.2 yet, so there might be other issues too.
Could you create a PR with this fix for version `2.2.1`?

**Comment by umarch06** [comment]
@jalogut were you able to fix it for magento 2.2.2?

**Comment by jalogut** [comment]
Hi @VincentMarmiesse and @umarch06 

Unfortunately I did not find the time to fix these issues for Magento 2.2 yet. I am quite busy at the moment, so I will not be able to fix it soon. A Pull request will be appreciated here.

**Comment by jalogut** [comment]
Hi @VincentMarmiesse and @umarch06 

`crud` template issue is now fixed on version 1.10. It was actually a small bug. Nothing to do with the Magento version.
Anyway I tested all templates and they are working up to Magento 2.2.3
